lib/mongoose_push/metrics/telemetry_metrics.ex

```diff
@@ -0,0 +1,49 @@
+defmodule MongoosePush.Metrics.TelemetryMetrics do
+  @moduledoc """
+  Module responsible for updating `Telemetry` metrics
+  """
+
+  def child_spec(_) do
+    TelemetryMetricsPrometheus.Core.child_spec(metrics: metrics())
+  end
+
+  def metrics do
+    [
+      # Summary is not yet supported in TelemetryMetricsPrometheus
+      Telemetry.Metrics.distribution("mongoose_push.notification.send.time",
+        buckets: [100, 250, 500, 1000],
+        tags: [:status, :service, :error_category, :error_reason],
+        tag_values: fn metadata ->
+          case metadata.status do
+            :success ->
+              Map.merge(
+                %{
+                  error_category: nil,
+                  error_reason: nil
+                },
+                metadata
+              )
+
+            :error ->
+              Map.merge(
+                %{
+                  error_category: Map.get(metadata, :type) || :generic,
+                  error_reason: metadata.reason
+                },
+                metadata
+              )
+          end
+        end
```

Comment:
Can these `tag_values` be prepared before the `Telemetry.Metrics.distribution` call?